src/cmds/hardware/lshw/lshw.c

```diff
@@ -0,0 +1,441 @@
+#include <net/inetdevice.h>
+#include <net/netdevice.h>
+#include <net/util/macaddr.h>
+#include <netinet/in.h>
+
+static void print_usage(void) {
+	printf("Usage: lshw [-h]\n");
+	printf("\t[-h]      - print this help\n");
+}
+
+static void print_error(void) {
+	printf("Wrong parameters\n");
+	print_usage();
+}
+
+/* lsusb implementation */
+static void show_usb_dev(struct usb_dev *usb_dev) {
+	printf("Bus %03d Device %03d ID %04x:%04x\n",
+			usb_dev->bus_idx,
+			usb_dev->addr,
+			usb_dev->dev_desc.id_vendor,
+			usb_dev->dev_desc.id_product
+	);
+}
+
+static void show_usb_desc_device(struct usb_dev *usb_dev) {
+	printf(
+			" Device Descriptor:\n"
+			"   b_length             %5u\n"
+			"   b_desc_type          %5u\n"
+			"   bcd_usb              %2x.%02x\n"
+			"   b_dev_class          %5u\n"
+			"   b_dev_subclass       %5u\n"
+			"   b_dev_protocol       %5u\n"
+			"   b_max_packet_size    %5u\n"
+			"   id_vendor           0x%04x\n"
+			"   id_product          0x%04x \n"
+			"   bcd_device           %2x.%02x\n"
+			"   i_manufacter         %5u\n"
+			"   i_product            %5u\n"
+			"   i_serial_number      %5u\n"
+			"   b_num_configurations %5u\n\n",
+			usb_dev->dev_desc.b_length, 
+			usb_dev->dev_desc.b_desc_type,
+			usb_dev->dev_desc.bcd_usb >> 8, usb_dev->dev_desc.bcd_usb & 0xff,
+			usb_dev->dev_desc.b_dev_class,
+			usb_dev->dev_desc.b_dev_subclass,
+			usb_dev->dev_desc.b_dev_protocol, 
+			usb_dev->dev_desc.b_max_packet_size,
+			usb_dev->dev_desc.id_vendor, 
+			usb_dev->dev_desc.id_product,
+			usb_dev->dev_desc.bcd_device >> 8, usb_dev->dev_desc.bcd_device & 0xff,
+			usb_dev->dev_desc.i_manufacter,
+			usb_dev->dev_desc.i_product,
+			usb_dev->dev_desc.i_serial_number,
+			usb_dev->dev_desc.b_num_configurations
+	);
+}
+
+static void show_usb_desc_interface(struct usb_dev *usb_dev) {
+	if (!usb_dev->usb_iface[0]->iface_desc[0]) {
+		printf(" Interface Descriptor:\n"
+			   "   No interfaces\n"
+		);
+		return;
+	}
+
+	printf(
+			" Interface Descriptor:\n"
+			"   b_length             %5u\n"
+			"   b_desc_type          %5u\n"
+			"   b_interface_number   %5u\n"
+			"   b_alternate_setting  %5u\n"
+			"   b_num_endpoints      %5u\n"
+			"   b_interface_class    %5u\n"
+			"   b_interface_subclass %5u\n"
+			"   b_interface_protocol %5u\n"
+			"   i_interface          %5u\n",
+			usb_dev->usb_iface[0]->iface_desc[0]->b_length,
+			usb_dev->usb_iface[0]->iface_desc[0]->b_desc_type,
+			usb_dev->usb_iface[0]->iface_desc[0]->b_interface_number,
+			usb_dev->usb_iface[0]->iface_desc[0]->b_alternate_setting,
+			usb_dev->usb_iface[0]->iface_desc[0]->b_num_endpoints,
+			usb_dev->usb_iface[0]->iface_desc[0]->b_interface_class,
+			usb_dev->usb_iface[0]->iface_desc[0]->b_interface_subclass,
+			usb_dev->usb_iface[0]->iface_desc[0]->b_interface_protocol,
+			usb_dev->usb_iface[0]->iface_desc[0]->i_interface
+	);
+}
+
+static void show_usb_desc_configuration(struct usb_dev *usb_dev) {
+	struct usb_desc_configuration *config = \
```

Comment:
Why do you use '\' here?

Reply:
I didn't get your point. This is the exact code, I didn't change anything. But the backslash can be removed.

Reply:
Oh, sorry. I didn't understand that it is a cloned code. But in any way, if you use code, it is yours